src/Mollie/WC/Plugin.php

```diff
@@ -440,43 +440,69 @@ public static function enqueueComponentsAssets()
     }
 
     /**
-     * Payment return url callback
+     * Returns the order from the Request first by Id, if not by Key
+     *
+     * @return bool|WC_Order
      */
-    public static function onMollieReturn ()
+    public static function orderByRequest()
     {
-        $dataHelper = self::getDataHelper();
+        $dataHelper = getDataHelper();
 
         $orderId = filter_input(INPUT_GET, 'order_id', FILTER_SANITIZE_NUMBER_INT) ?: null;
         $key = filter_input(INPUT_GET, 'key', FILTER_SANITIZE_STRING) ?: null;
         $order = $dataHelper->getWcOrder($orderId);
 
         if (!$order) {
-            self::setHttpResponseCode(404);
-            self::debug(__METHOD__ . ":  Could not find order $orderId.");
-            return;
+            $order = $dataHelper->getWcOrder(wc_get_order_id_by_order_key ($key));
+        }
+
+        if (!$order) {
+            throw new \RuntimeException(
+                "Could not find order by order Id {$orderId}",
+                404
+            );
         }
 
         if (!$order->key_is_valid($key)) {
-            self::setHttpResponseCode(401);
-            self::debug(__METHOD__ . ":  Invalid key $key for order $orderId.");
+            throw new \RuntimeException(
+                "Invalid key given. Key {$key} does not match the order id: {$orderId}",
+                401
+            );
+        }
+
+        return $order;
+    }
+    /**
+     * Payment return url callback
+     */
+    public static function onMollieReturn ()
+    {
+        $dataHelper = getDataHelper();
+
+        try {
+            $order = self::orderByRequest();
+        } catch(RuntimeException $exc) {
+            self::setHttpResponseCode($exc->getCode());
+            debug(__METHOD__ . $exc->getMessage());
```

Comment:
Missing space in string between `__METHOD__` and the exception message.
Dunno if all of the debug message in mollie have a `:` after the method name I saw some of them are having it but not sure if all of them, would be nice to have a single format for the message.